# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditorBase.cpp
# Commit: 695d708318a5
# Full Hash: 695d708318a576da278e5c035f77503274e49b7b
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-08-04 09:46:07
# Regressor Bug: 1774704
# File Overlap Count: 1
# Description:
#   Bug 1774704 - part 5-3: Make `HTMLEditor::HandleCSSIndentAtSelectionInternal` stop touching `Selection` directly and setting `mNewBlockElement` r=m_kato
#   
#   It's called only by `HTMLEditor::HandleCSSIndentAtSelection` which is called
#   only by `HTMLEditor::HandleIndentAtSelection`.  They don't touch `Selection`
#   after calling it.  Therefore, we can make it adjust collapsing selection point
# ==============================================================================

diff -r 80ed14c716a2 -r 695d708318a5 editor/libeditor/EditorBase.cpp
--- a/editor/libeditor/EditorBase.cpp	Thu Aug 04 03:00:31 2022 +0000
+++ b/editor/libeditor/EditorBase.cpp	Thu Aug 04 03:16:28 2022 +0000
@@ -5995,6 +5995,7 @@
 bool EditorBase::IsSelectionRangeContainerNotContent() const {
   MOZ_ASSERT(IsEditActionDataAvailable());
 
+  // TODO: Make all callers use !AutoRangeArray::IsInContent() instead.
   const uint32_t rangeCount = SelectionRef().RangeCount();
   for (const uint32_t i : IntegerRange(rangeCount)) {
     MOZ_ASSERT(SelectionRef().RangeCount() == rangeCount);